services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/dataaccess/AccessRequirementManagerImpl.java

```diff
@@ -119,6 +132,13 @@ public static void populateModifiedFields(UserInfo userInfo, AccessRequirement a
 			throw new UnauthorizedException("Only ACT member can create an AccessRequirement.");
 		}
 		populateCreationFields(userInfo, accessRequirement);
+		
+		for (RestrictableObjectDescriptor rod : accessRequirement.getSubjectIds()) {
```

Comment:
We should probably do the same check when we update an AR, e.g. when the list of restrictable objects are different